### ISSUE TITLE
CR Request for Web of Things (WoT) Discovery

### ISSUE THREAD

**Issue by mmccool** [w3c/transitions#465]
# Document title, URLs, estimated publication date
Web of Things (WoT) Discovery
[Static CR snapshot: https://github.com/w3c/wot-discovery/blob/main/publication/4-cr/Overview.html](https://github.com/w3c/wot-discovery/blob/main/publication/4-cr/Overview.html)
Publication as soon as possible after October 26 (actual date needs to be updated in document)

# Abstract
The W3C Web of Things (WoT) is intended to enable interoperability across IoT platforms and application domains. One key mechanism for accomplishing this goal is the definition and use of metadata describing the interactions an IoT device or service makes available over the network at a suitable level of abstraction. The WoT Thing Description specification satisfies this objective.

However, in order to use a Thing its Thing Description first has to be obtained. The WoT Discovery process described in this document addresses this problem. WoT Discovery needs to support the distribution of WoT Thing Descriptions in a variety of use cases. This includes ad-hoc and engineered systems; during development and at runtime; and on both local and global networks. The process also needs to work with existing discovery mechanisms, be secure, protect private information, and be able to efficiently handle updates to WoT Thing Descriptions and the dynamic and diverse nature of the IoT ecosystem.

The WoT Discovery process is divided into two phases, Introduction, and Exploration. The Introduction phase leverages existing discovery mechanisms but does not directly expose metadata; they are simply used to discover Exploration services, which provide metadata but only after secure authentication and authorization. This document normatively defines two Exploration services: for distributing a single WoT Thing Description from a regular web service, including as a special case self-description; and a searchable WoT Thing Description Directory service for collections of Thing Descriptions. A variety of Introduction services are also described and where necessary normative definitions are given to support them.

# Status
[Status of the document (https://github.com/w3c/wot-discovery/blob/main/publication/4-cr/Overview.html#sotd)](https://github.com/w3c/wot-discovery/blob/main/publication/4-cr/Overview.html#sotd)

# Link to group's decision to request transition
[Resolution to request CR transition for WoT Discovery in WG call October 26.](https://www.w3.org/2022/10/26-wot-minutes.html#r02)

# Changes
Changes from 10 August 2022 Working Draft
- Clarify 3x rule for DDoS migitation to allow padding in request, and require it only outside of protected local networks.
- Define content types to be used for CoAP servers, paralleling TD servers.
- Update DID service names and refine DID assertions and add examples to clarify use for Introductions.
- Update DNS-SD section to include explicit scheme in TXT field.
- Replace reference to draft CoRE-RD with final specification published as RFC9176.

# Requirements satisfied
Requirements laid out in the WG charter for this deliverable were satisfied except for geolocation, which was deferred to future work.

# Dependencies met (or not)
- We are still waiting for registration of a DNS-SD service name.  An ednote has been placed in the document regarding this.  If a service name cannot be granted, this section will be removed in its entirety.  If a different name is necessary it will not change the functionality of this feature.
- DID service names are yet to be registered (this is an internal W3C registry, not an IANA registry)

# Wide Review
- Wide review was requested from Accessibility, Internationalization, Security, Privacy, and TAG and all issues raised addressed (although in some cases we have not yet gotten confirmation that the issues were addressed from the reviewers):
    - https://github.com/w3cping/privacy-request/issues/83
    - https://github.com/w3c/security-request/issues/25
    - https://github.com/w3c/a11y-request/issues/24
    - https://github.com/w3c/i18n-request/issues/175
    - https://github.com/w3ctag/design-reviews/issues/733
- The Societal Impact questionnaire has not yet been filled in but will be completed prior to PR.

# Issues addressed
- General issue tracker: https://github.com/w3c/wot-discovery/issues
- Issues marked [resolve by PR](https://github.com/w3c/wot-discovery/labels/resolve%20by%20PR) will be addressed prior to PR transition.  These are editorial, except for the above noted IANA registration for DNS-SD which is pending (see [Issue 403](https://github.com/w3c/wot-discovery/issues/403))
- All other open issues will be deferred to future work (e.g. in a new charter)

# Formal Objections
None at present.

# Implementation
A DRAFT implementation report is available at [https://w3c.github.io/wot-discovery/testing/report.html](https://w3c.github.io/wot-discovery/testing/report.html).   This will be updated during the CR review period to address
at-risk items and add testimonials.

# Patent disclosures
None.

**Comment by plehegar** [comment]
* [Horizontal issues](https://www.w3.org/PM/horizontal/review.html?shortname=wot-discovery)
* [Normative references](https://labs.w3.org/normative-references/check?url=https://w3c.github.io/wot-discovery/publication/4-cr/Overview.html)

**Comment by swickr** [comment]
CR Exit criteria are missing.

#320 appears to be in active discussion.

**Comment by mmccool** [comment]
CR Exit criteria have been added to the CR candidate in the SOTD section, the text is as follows:
```
    The Web of Things Working Group intends to submit this document for
    consideration as a W3C Proposed Recommendation after at least the
    minimum CR review period has passed.
    However, before PR transition is requested,
    any features or assertions currently marked as at-risk that do not
    have at least two implementations at that time will either be
    removed or converted into informative statements, as appropriate.
```

Also, we made sure the list of the at-risk assertions in the SOTD is complete and exhaustive.

There are indeed some normative reference issues flagged by ReSpec (e.g. normative references to informative documents) but note these are in reference to internationalization glossary entries and are part of assertions supporting internationalization.  Please advise how to resolve.

#320 is going on in parallel.  If DID does NOT get approved, we can remove this "Introduction" mechanism.  Should we add a note saying this section is at risk and depends on the DID spec reaching PR status?

Regarding the horizontal issues we believe these have been resolved, but have been waiting for closure from the i18y and security groups.

**Comment by mmccool** [comment]
Also, regarding https://github.com/w3c/wot-discovery/issues/254, which is one of the "Horizontal Issues" that show up on https://www.w3.org/PM/horizontal/review.html?shortname=wot-discovery, this one had the "security-needs-resolution" label added to it by accident.  Both myself and PLH have tried to remove this label but the w3cbot keeps putting it back.  So this is not a "real" open horizontal review issue, it's a tooling problem.

**Comment by sebastiankb** [comment]
@swickr are there news about the Discovery publications?

**Comment by mmccool** [comment]
Note that there are two similar incorrectly labelled issues in WoT Architecture.

**Comment by mmccool** [comment]
Links have been updated as described above by the following PR: https://github.com/w3c/wot-discovery/pull/444

Internationalization issues have all been resolved in discussion with i18n group.  Some changes were needed to convert "UTF-8 lexicographic order" to the more correct "Unicode code point order": https://github.com/w3c/wot-discovery/pull/441 and https://github.com/w3c/wot-discovery/pull/443

At this point we feel the horizontal review and normative references issues have both been addressed.

**Comment by mmccool** [comment]
@swickr @plehegar @ylafon  We believe we have resolved the above issues and are ready for CR transition.  According to the W3C procedure are just waiting for one of you to do a final review and give approval.

**Comment by mmccool** [comment]
The CR draft has been updated to use HTML rather than EVENTSOURCE and we had a resolution in the WoT WG to make this change.  Therefore we can proceed with CR publication.

**Comment by plehegar** [comment]
https://www.w3.org/TR/2023/CR-wot-discovery-20230119/
